posthog/models/surveys/util.py

```diff
@@ -0,0 +1,36 @@
+def get_survey_response_clickhouse_query(question_index: int, question_id: str | None = None) -> str:
+    """
+    Generate a ClickHouse query to extract survey response based on question index or ID
+
+    Args:
+        question_index: Index of the question
+        question_id: ID of the question (optional)
+
+    Returns:
+        The survey response or empty string if not found
+    """
+    id_based_key = _build_id_based_key(question_index, question_id)
+    index_based_key = _build_index_based_key(question_index)
+
+    return _build_coalesce_query(id_based_key, index_based_key)
+
+
+def _build_id_based_key(question_index: int, question_id: str | None = None) -> str:
+    if question_id:
+        return f"'$survey_response_{question_id}'"
```

Comment:
`survey_response = '$survey_response'` so we can reuse this const everywhere else